asv_bench/benchmarks/groupby.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+import xarray as xr
+
+from . import parameterized, requires_dask
+
+
+class GroupBy:
+    def setup(self, *args, **kwargs):
+        self.ds = xr.Dataset(
+            {
+                "a": xr.DataArray(np.r_[np.arange(500.0), np.arange(500.0)]),
+                "b": xr.DataArray(np.arange(1000.0)),
+            }
+        )
+
+    @parameterized(["method"], [("sum", "mean")])
+    def agg(self, method):
+        return getattr(self.ds.groupby("a"), method)()
+
+
+class GroupByDask(GroupBy):
+    def setup(self, *args, **kwargs):
+        requires_dask()
+        super().setup(**kwargs)
+        self.ds = self.ds.chunk({"dim_0": 50})
+
+
+class GroupByDataFrame(GroupBy):
```

Comment:
I'm not sure we really need the dataframe tests.

Reply:
I think it's good to have some kind of "competitor" comparison, because performance complaints will always be relative to them anyway. It might not be needed for more than just a few of the (future) tests though.